src/llm_chat.ts

```diff
@@ -814,8 +814,8 @@ export class LLMChatPipeline {
     let prompts;
     // beginning of the conversation
     if (this.filledKVCacheLength === 0) {
-      if (this.conversation.config.add_bos) {
-        tokens = [this.bosTokenId];
+      if (this.conversation.config.system_prefix_token_ids !== undefined) {
+        tokens = this.conversation.config.system_prefix_token_ids;
```

Comment:
```suggestion
        tokens = [...this.conversation.config.system_prefix_token_ids];
```

Just one small nit on making a deep copy for `system_prefix_token_ids`. I found out this because assertions in `seed.ts` were not able to pass; after printing I realized we are actually mutating the `config.system_prefix_token_ids`, leading to future runs pretending with extra `tokens`.